core/chains/evm/gas/block_history_estimator.go

```diff
@@ -838,20 +838,23 @@ func (b *BlockHistoryEstimator) EffectiveGasPrice(block evmtypes.Block, tx evmty
 			b.logger.Warnw("Got transaction type 0x2 but one of the required EIP1559 fields was missing, falling back to gasPrice", "block", block, "tx", tx)
 			return tx.GasPrice
 		}
-		if tx.MaxFeePerGas.Cmp(block.BaseFeePerGas) < 0 {
-			// This should not pass config validation
+		assumptionViolation := false
+		if tx.MaxFeePerGas.Cmp(block.BaseFeePerGas) <= 0 {
 			b.logger.AssumptionViolationw("MaxFeePerGas >= BaseFeePerGas", "block", block, "tx", tx)
-			return nil
+			assumptionViolation = true
 		}
-		if tx.MaxFeePerGas.Cmp(tx.MaxPriorityFeePerGas) < 0 {
-			// This should not pass config validation
+		if tx.MaxFeePerGas.Cmp(tx.MaxPriorityFeePerGas) <= 0 {
 			b.logger.AssumptionViolationw("MaxFeePerGas >= MaxPriorityFeePerGas", "block", block, "tx", tx)
-			return nil
+			assumptionViolation = true
```

Comment:
As well - if those assumptions are violated and `tx.GasPrice` is nil, the ending effective gas price calculations will return values that are an underestimation in 2/3 cases:

Case 1: Base > Max, Max > Max priority: returns Max. underestimation since in actuality the tx gas is probably greater than Base
Case 2: Base > Max, Max priority > Max: returns Max. underestimation since in actuality the tx gas is probably greater than Base.
Case 3: Base < Max, Max priority > Max: returns Max. correct for estimation since in actuality the tx gas is less than or equal to the Max.